app/src/main/java/com/concordia/mcga/models/Floor.java

```diff
@@ -65,6 +89,33 @@ public void setBuilding(Building building) {
         return escalators;
     }
 
+    public void addElevator(Elevator elevator) {
+        elevators.add(elevator);
+    }
+
+    public void addStaircase(Staircase staircase) {
+        staircases.add(staircase);
+    }
+
+    public void addEscalator(Escalator escalator) {
+        escalators.add(escalator);
+    }
+
+    public void populateTiledMap() {
+        TiledMap map = null;
+        try {
+            map = IndoorMapFactory.getInstance().createTiledMap(building, floorNumber);
+            IndoorMapFactory.getInstance().insertWalkablePaths(building, floorNumber, map);
+        } catch (MCGADatabaseException e) {
+            e.printStackTrace();
```

Comment:
we should propagate errors found here. Else NullPointerExceptions will bite us when manipulating the map